python/simulation.py

```diff
@@ -0,0 +1,689 @@
+from __future__ import division, print_function
+
+import numbers
+import os
+import re
+import sys
+
+import meep as mp
+from meep.geom import Vector3
+from meep.source import EigenModeSource, check_positive
+
+# TODO(chogan): Hopefully we don't have to require funcsigs
+try:
+    from inspect import signature
+except ImportError:
+    from funcsigs import signature
+
+CYLINDRICAL = -2
+no_size = 1e-20
+
+
+def get_num_args(func):
+    sig = signature(func)
```

Comment:
`signature` takes `self` into account and doesn't include it since it's a bound argument. I've removed the functools dependency by doing this for python 2
```python
def get_num_args(func):
    argc = func.func_code.co_argcount
    if 'self' in func.func_code.co_varnames:
        argc -= 1
    return argc
```